src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs

```diff
@@ -25,30 +23,61 @@ public class ServiceManagerOptions
 
         /// <summary>
         /// Gets or sets the connection string of Azure SignalR Service instance.
+        /// Gets or sets the connection string of Azure SignalR Service instance and switches to single-endpoint mode.
         /// </summary>
-        public string ConnectionString { get; set; } = null;
+        public string ConnectionString
+        {
+            get => _connectionString;
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    throw new ArgumentException($"'{nameof(ConnectionString)}' cannot be null or whitespace", nameof(ConnectionString));
+                }
+                else
+                {
+                    _connectionString = value;
+                    ServiceEndpoint = new ServiceEndpoint(ConnectionString);
+                }
+            }
+        }
 
         /// <summary>
         /// Gets or sets the proxy used when ServiceManager will attempt to connect to Azure SignalR Service.
         /// </summary>
         public IWebProxy Proxy { get; set; }
 
         /// <summary>
-        /// Gets or sets the service endpoint for accessing Azure SignalR Service.
+        /// Gets or sets the service endpoint for accessing Azure SignalR Service and switches to single-endpoint mode.
         /// </summary>
         public ServiceEndpoint ServiceEndpoint
         {
-            get
+            get => _serviceEndpoint;
+            set
             {
-                if (_serviceEndpoint == null)
-                {
-                    _serviceEndpoint = new ServiceEndpoint(ConnectionString, EndpointType.Secondary);
-                }
-                return _serviceEndpoint;
+                _multiEndpointState = false;
+                _serviceEndpoint = value;
+                _serviceEndpoints = null;
```

Comment:
with this impl, order matters, e.g.

```cs
var options = new ServiceManagerOptions{
   ServiceEndpoint = ...
   ServiceEndpoints = ...
}

```
is different than 

```cs
var options = new ServiceManagerOptions{
   ServiceEndpoints = ...
   ServiceEndpoint = ...
}

```

How about validating before using the option that these properties should not be set at the same time?